Apps/Alarm_controller/alarm_controller.c

```diff
@@ -42,90 +43,69 @@ void AlarmController_Init(void)
  ******************************************************************************/
 void AlarmController_Loop(void)
 {
-    static short previous_id       = -1;
-    static uint8_t blink           = 0;
-    static uint8_t blink_nb        = BLINK_NUMBER * 2;
-    static uint32_t last_blink     = 0;
-    static uint32_t detection_date = 0;
+    static short previous_id   = -1;
+    static uint8_t blink       = 0;
+    static uint8_t blink_nb    = BLINK_NUMBER * 2;
+    static uint32_t last_blink = 0;
 
     // ********** hot plug management ************
     // Check if we have done the first init or if service Id have changed
-    if (previous_id != RoutingTB_IDFromService(app))
+    if (Luos_IsNodeDetected())
     {
-        if (RoutingTB_IDFromService(app) == 0)
+        if (previous_id == -1)
```

Comment:
Why do you have to keep this test ?

Reply:
again it is used to keep that the detection status before and after a new detection. I will change it with the end detection message like the gate